lbrynet/core/client/ClientProtocol.py

```diff
@@ -208,10 +208,11 @@ def get_next_request(results):
                         log.info(result.value)
                         log.info("The connection to %s is closing due to an error: %s",
                                  self.peer, result.getTraceback())
+
+                        self.peer.report_down()
             if failed is False:
                 log.debug("Asking for another request from %s", self.peer)
-                from twisted.internet import reactor
-                reactor.callLater(0, self._ask_for_request)
+                self._ask_for_request()
```

Comment:
Looking through the code again - you're right. I got confused and thought there was a loop between ask_for_request and handle_request.